aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/AWSS3StoragePlugin.java

```diff
@@ -138,7 +161,7 @@ public void configure(@NonNull JSONObject pluginConfiguration, @NonNull Context
     @NonNull
     @Override
     public AmazonS3Client getEscapeHatch() {
-        return storageService.getClient();
+        return ((AWSS3StorageService) storageService).getClient();
```

Comment:
Our `StorageService` is already scoped into this S3 plugin, so I think it'd be fair to just bake an S3 gizmo right into it. Maybe we should use a non-`final` `AWSS3StorageService`?

Reply:
I personally dont even like the fact that `StorageService` directly interacts with `TransferObserver`. Considering that our ultimate goal is to not rely on `TransferUtility`, we should look into re-factoring it in the near future